apps/builder/app/builder/features/ai/ai-command-bar.tsx

```diff
@@ -247,51 +159,89 @@ export const AiCommandBar = () => {
           <ScrollArea css={{ maxHeight: theme.spacing[29] }}>
             <AutogrowTextArea
               autoFocus
-              disabled={textAreaDisabled}
-              placeholder={textAreaPlaceholder}
-              value={textAreaValue}
-              onChange={setValue}
-              onKeyDown={(event) => {
+              disabled={status !== "idle"}
+              placeholder={
+                status === "recording"
+                  ? "Recording..."
+                  : status === "transcribing"
+                  ? "Transcribing recording..."
+                  : "Enter value..."
+              }
```

Comment:
Are you think I moved all this `? :` into linear imperative `if's` for fun ;-) The code above is unsupportable. Every new state would make this grow grow grow. See in previous code, new state just single `if` in render was added

Reply:
move it to a function that resolves the state, then its easier to read, even better if its a state machine